crates/biome_graphql_analyze/src/lint/nursery/no_anonymous_operations.rs

```diff
@@ -0,0 +1,116 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_graphql_factory::make;
+use biome_graphql_syntax::{GraphqlOperationDefinition, GraphqlOperationType};
+use biome_rowan::{AstNode, BatchMutationExt};
+use biome_string_case::Case;
+
+use crate::GraphqlRuleAction;
+
+declare_lint_rule! {
+    /// Require specifying name for GraphQL operations.
+    ///
+    /// This is useful since most GraphQL client libraries are using the operation name for caching purposes.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```graphql,expect_diagnostic
+    /// query {}
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```graphql
+    /// query Human {
+    ///   name
+    /// }
+    /// ```
+    ///
+    pub NoAnonymousOperations {
+        version: "next",
+        name: "noAnonymousOperations",
```

Comment:
I read the documentation, and it seems the rule is more focused on the name of the operations. It doesn't say anything about operations that don't have a name.

Wouldn't it make more sense to call the rule `useNamedOperation`? Plus, let's use the singular